FHEM/14_SD_Keeloq.pm

```diff
@@ -216,18 +217,26 @@ sub Attr(@) {
 					$attr{$name}{ChannelNames} = $attrValue;
 				}
 
-				if ($attrName eq "Channels" && $attrValue == 0 && $addGroups eq "") {
-					return "ERROR: you can use Channels = $attrValue only with defined attribut addGroups!";
+				if ($attrName eq "Channels") {
+					my $Channels = AttrVal($name, "Channels", "");
+					return "ERROR: you can use Channels = $attrValue only with defined attribut addGroups!" if ($attrValue == 0 && $addGroups eq "");
+					if ($Channels > $attrValue) {
+						foreach my $d (sort keys %{$hash->{READINGS}}) {
```

Comment:
Ist das sortieren notwendig?

Reply:
In der Sache nicht unbedingt. Es wäre ein klassischer Copy&Paste wie dies da hinkam :-)

Würde das zusätzliche sort so viel ausmachen an Ressourcen? Ja es wäre unötige aber rein informell mal.

Reply:
Es verbraucht halt CPU Ressourcen weil die komplette Liste sortiert wird.
Wenn man es nicht braucht, sollte man es weglassen

Reply:
Es ist doch schon geschehen :-) Somit ist nur das benötigte im Code.